### ISSUE TITLE
generate jacoco report failed due to "Error while analyzing XXXX-classes"

### ISSUE THREAD

**Issue by siemens84cn** [jacoco/jacoco#1276]
error logs in eclipse tool console:
begin to generate coverage report

```
java.io.IOException: Error while analyzing /tmp/qc-service/BOOT-INF/classes/com/xxx/dd/seq/common/config/TimeZoneConfig.class.
coverage report generate successfully!
	at org.jacoco.core.analysis.Analyzer.analyzerError(Analyzer.java:162)
	at org.jacoco.core.analysis.Analyzer.analyzeClass(Analyzer.java:134)
	at org.jacoco.core.analysis.Analyzer.analyzeClass(Analyzer.java:157)
	at org.jacoco.core.analysis.Analyzer.analyzeAll(Analyzer.java:193)
	at org.jacoco.core.analysis.Analyzer.analyzeAll(Analyzer.java:226)
	at org.jacoco.core.analysis.Analyzer.analyzeAll(Analyzer.java:221)
	at org.jacoco.core.analysis.Analyzer.analyzeAll(Analyzer.java:221)
	at org.jacoco.core.analysis.Analyzer.analyzeAll(Analyzer.java:221)
	at org.jacoco.core.analysis.Analyzer.analyzeAll(Analyzer.java:221)
	at org.jacoco.core.analysis.Analyzer.analyzeAll(Analyzer.java:221)
	at org.jacoco.core.analysis.Analyzer.analyzeAll(Analyzer.java:221)
	at org.jacoco.core.analysis.Analyzer.analyzeAll(Analyzer.java:221)
	at com.xxx.dd.seq.coverage.client.JacocoClient.analyzeStructure(JacocoClient.java:158)
	at com.xxx.dd.seq.coverage.client.JacocoClient.ReportGenerator(JacocoClient.java:94)
	at com.xxx.dd.seq.coverage.service.Impl.TaskServiceImpl.runTask(TaskServiceImpl.java:57)
	at com.xxx.dd.seq.coverage.controller.TaskController.lambda$trigger$0(TaskController.java:40)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalArgumentException
	at org.objectweb.asm.ClassVisitor.<init>(ClassVisitor.java:75)
	at org.jacoco.core.internal.flow.ClassProbesVisitor.<init>(ClassProbesVisitor.java:38)
	at org.jacoco.core.internal.flow.ClassProbesVisitor.<init>(ClassProbesVisitor.java:28)
	at org.jacoco.core.internal.analysis.ClassAnalyzer.<init>(ClassAnalyzer.java:59)
	at org.jacoco.core.analysis.Analyzer$1.<init>(Analyzer.java:97)
	at org.jacoco.core.analysis.Analyzer.createAnalyzingVisitor(Analyzer.java:94)
	at org.jacoco.core.analysis.Analyzer.analyzeClass(Analyzer.java:114)
	at org.jacoco.core.analysis.Analyzer.analyzeClass(Analyzer.java:132)
	... 17 more
```

jacoco plugin version is as follows:

```
		<dependency>
		    <groupId>org.jacoco</groupId>
		    <artifactId>org.jacoco.core</artifactId>
		    <version>0.8.7</version>
		</dependency>
		<!-- https://mvnrepository.com/artifact/org.jacoco/jacoco-maven-plugin -->
		<dependency>
		    <groupId>org.jacoco</groupId>
		    <artifactId>jacoco-maven-plugin</artifactId>
		    <version>0.8.7</version>
		</dependency>
```

on the other hand, i used jacoco command line to generate report, it's successed by the same parameters.

java -jar /Users/siemens/Desktop/jacoco-0.8.7/lib/jacococli.jar report /tmp/jacoco.exec --html jacoco_html --classfiles /tmp/qc-service/BOOT-INF/classes --sourcefiles /Users/siemens/eclipse-workspace/qc-service/src

Anyone can help on this issue?
https://groups.google.com/forum/?fromgroups=#!forum/jacoco

Thank you!

**Comment by Godin** [comment]
Your stack-trace of failed attempt contains

> at com.xxx.dd.seq.coverage.client.JacocoClient.analyzeStructure(JacocoClient.java:158)

**which is not something developed by us**, 
and since there is no failure when using our `jacococli.jar`,
you should talk with developers of `com.xxx.dd.seq.coverage.client.JacocoClient`:

because of

> Caused by: java.lang.IllegalArgumentException
> at org.objectweb.asm.ClassVisitor.(ClassVisitor.java:75)

seems that version of ASM library used by `com.xxx.dd.seq.coverage.client.JacocoClient` doesn't match version required by JaCoCo - see https://gitlab.ow2.org/asm/asm/-/blob/ASM_7_3_1/asm/src/main/java/org/objectweb/asm/ClassVisitor.java#L74

**Comment by siemens84cn** [comment]
Hi

> Your stack-trace of failed attempt contains
> 
> > at com.xxx.dd.seq.coverage.client.JacocoClient.analyzeStructure(JacocoClient.java:158)
> 
> **which is not something developed by us**, and since there is no failure when using our `jacococli.jar`, you should talk with developers of `com.xxx.dd.seq.coverage.client.JacocoClient`:
> 
> because of
> 
> > Caused by: java.lang.IllegalArgumentException
> > at org.objectweb.asm.ClassVisitor.(ClassVisitor.java:75)
> 
> seems that version of ASM library used by `com.xxx.dd.seq.coverage.client.JacocoClient` doesn't match version required by JaCoCo - see https://gitlab.ow2.org/asm/asm/-/blob/ASM_7_3_1/asm/src/main/java/org/objectweb/asm/ClassVisitor.java#L74

Hi,Godin

i just fixed the issue because of org.ow2.asm jar package conflict. the old one is 6.0, the latest one is 9.0.
Once i omitted the old one, it can generate coverage report as normal.

Thanks for your help!

**Comment by Godin** [comment]
> org.ow2.asm jar package conflict. the old one is 6.0, the latest one is 9.0.

The latest released as of today is actually `9.2` - see https://search.maven.org/artifact/org.ow2.asm/asm
JaCoCo `0.8.7` actually requires `9.1` - see https://www.jacoco.org/jacoco/trunk/doc/changes.html
